sdk/communication/communication-jobrouter/src/routerClient.ts

```diff
@@ -0,0 +1,426 @@
+   * `@azure/identity` or TokenCredential to create a credential.
+   * @param routerClientOptions - Optional. Options to configure the HTTP pipeline.
+   */
+  constructor(
+    endpoint: string,
+    credential: KeyCredential | TokenCredential,
+    routerClientOptions?: RouterClientOptions
+  );
+
+  /**
+   * Initializes a new instance of the RouterClient class using a TokenCredential.
+   * @param endpoint - The endpoint of the service (ex: https://contoso.eastus.communications.azure.net).
+   * @param credential - CommunicationTokenCredential that is used to authenticate requests to the service.
+   * @param routerClientOptions - Optional. Options to configure the HTTP pipeline.
+   */
+  constructor(
+    endpoint: string,
+    credential: CommunicationTokenCredential,
+    routerClientOptions?: RouterClientOptions
+  );
+
+  /**
+   * Creates an instance of the RouterClient for a given resource and user.
+   *
+   * @param connectionStringOrUrl - The connectionString or url of the Communication Services resource.
+   * @param credentialOrOptions - The key or token credential or RouterClientOptions. Use AzureCommunicationKeyCredential from \@azure/communication-common to create a credential.
+   * @param maybeOptions - Additional client options.
+   */
+  constructor(
+    connectionStringOrUrl: string,
+    credentialOrOptions?:
+      | KeyCredential
+      | TokenCredential
+      | CommunicationTokenCredential
+      | RouterClientOptions,
+    maybeOptions: RouterClientOptions = {}
+  ) {
+    const { url, credential } = parseClientArguments(connectionStringOrUrl, credentialOrOptions);
+    const options = isRouterClientOptions(credentialOrOptions) ? credentialOrOptions : maybeOptions;
+
+    const libInfo = `azsdk-js-communication-jobrouter/${SDK_VERSION}`;
+
+    if (!options.userAgentOptions) {
+      options.userAgentOptions = {};
+    }
+
+    const userAgentOptions = { ...options.userAgentOptions };
+    if (options.userAgentOptions.userAgentPrefix) {
+      userAgentOptions.userAgentPrefix = `${options.userAgentOptions.userAgentPrefix} ${libInfo}`;
+    } else {
+      userAgentOptions.userAgentPrefix = libInfo;
+    }
+
+    const internalPipelineOptions: InternalPipelineOptions = {
+      ...options,
+      userAgentOptions,
+      loggingOptions: {
+        logger: logger.info
+      }
+    };
+
+    const authPolicy = createCommunicationAuthPolicy(credential);
+
+    this.client = new JobRouterApiClient(url, internalPipelineOptions);
+    this.client.pipeline.addPolicy(authPolicy);
+  }
+
+  // Job Actions
+  /**
+   * Creates a job.
+   * Returns job with the id of the created job.
+   * @param jobId - The job to be create
+   * @param options - Operation options.
+   */
+  public async createJob(
+    jobId: string,
+    options: CreateJobOptions = {}
+  ): Promise<RouterJobResponse> {
+    const jobModel = options;
+    const job = await this.client.jobRouter.upsertJob(jobId, jobModel, options);
```

Comment:
You will need to enable tracing. Maybe you can add it as a follow up PR, but ideally should be in for beta. This is an example of how tracing is integrated. If you decide to make this in a separate PR please file an issue to track it and add it as a comment here

https://github.com/Azure/azure-sdk-for-js/blob/295dbb025f646c6e846da877abda9e9d6b2ff8f8/sdk/template/template/src/configurationClient.ts#L143

Reply:
Created an issue for it, I will do it in a follow up PR.
https://github.com/Azure/azure-sdk-for-js/issues/23008